### ISSUE TITLE
Signal Percentage not accurate

### ISSUE THREAD

**Issue by ozone1864** [openatv/enigma2#1125]
Dear Captain,
I don't know if it's GUI or driver problem but the thing is 
Signal Percentage for channels is like 20% less the original values
Tested 3 receivers on same channel all showed 73% except zgemma
H9S is showing 55% on all channels.
Firmware OPENATV 6.2 Latest version
Regards,

**Comment by Tony-il-Capo** [comment]
That is probably hardware related(/driver limitation), some STBs are known to show strange signal levels, but it doesn't affect reception. (signal is the same, but somehow get measured differently)

Also signal figures in percentage are merely indicative, different hardware different % of signal, depends on how internal calculations are made.

If you don't have special measuring equipment, when you need to adjust your dish always use a STB that is known to report reliable signal in dB (you need to have an idea how much dB you need, standard (DVB/S2), modulation and  FEC is what dictates [minimum signal to lock](http://www.satbroadcasts.com/news,81,Minimum_carrier_to_noise_ratio_values_CNR,_CN_for_DVB_S2_system.html)).

**Comment by arn354** [comment]
https://github.com/openatv/enigma2/blob/6.3/lib/dvb/frontend.cpp#L1236
some tuners get a "special" treatment in frontend.cpp - if not recalculated there, enigma2 shows what the driver reports (correct or not). The shown number (whatever it is) should only be taken as an indicator not as a scientific value of professional measurement equipment. And due to that "special" treatment you should not compare Box A and Box B.
If Box B has a poorer reception then Box A, but Box B is recalculated - it is possible that Box B is showing a higher SNR value then Box A - indicating that Box B has a better reception. There is no normed comparison between the boxes.
If you propose a proper re-calculation for that box as pull-request here - your value will get cosmetically fixed.

**Comment by AbuBaniaz** [comment]
See here for more information about some related 

https://github.com/OpenViX/enigma2/issues/398

**Comment by EMJB** [comment]
This is a complex issue and an example of some of the points I tried to raise in the OpenVix thread. I have failed to find a definition of what this data should be, and it would appear that "fixes" to specific perceived problems in this area have been made by various people (myself included) based on an incomplete understanding of all the problems and/or misled by confusing terminology. Thus the problem of changing values raised by the OP do not surprise me, and I would expect further "accidental" changes unless/until a comprehensive review is made of the both the available input data and required outputs leading to a single co-ordinated update of frontend.cpp. In view of the issues associated with multiple images, potentially different views as to what information the PVR User should be given, specialised knowledge required, and the need to collect data on a large number of tuner types, this needs to be a team effort.

Before anyone attempts another "fix", I would like to highlight the following from the points I made in the OpenVix thread:

(1) The legacy API "snrValue" is so poorly defined that describing the results one should be saying they are "not what one would like" rather than "wrong". Fortunately the new API definition of the equivalent parameter is much better, and I have yet to see any firm evidence for the need for tuner-specific corrections or driver changes.

(2) The current processing in frontend.cpp appears to attempt to produce significantly different values for  signalQualitydB according to which API is used, with differences of ~20 dB, or similar to the expected range of values in some cases. Hence there will very significant differences between displayed signal percentages according to the API used (a marginal signal will be displayed as 0% using the old API data, an possibly as high as 50% using the new API).

(3) The code to make the necessary corrections to align the signalQualitydB data is relatively simple - I would guess somewhere in the range 1-20 lines of code, but obviously agreements needed as to which is to be changed before implementing that code. Unfortunately determining the precise values to be used in any correction logic to align the two versions are far from simple. I believe the figure of 20 dB is probably accurate within 2 or 3 dB for current UK DVB-T transmissions and I could improve on this with some further investigation, but lack both the theoretical background and the equipment to even start extrapolating this to DVB-S & DVB-C, and other DVB-T modulation characteristics. 

With respect to arn354's comments I would agree as statement of the current situation, but believe that comparability between tuners is achievable, and highly desirable. The comparability of the percentage figures between tuners depend just two factors, the consistency of the signalQualitydB data and the upper limit applied. I believe there is good reason to expect that resolving the issues within frontend.cpp coupled with the improved definition of the new API will resolve the former for at least the majority of tuners (including all new ones). At the moment the old API processing sometimes applies tuner-specific limits, but I have recommended this is abandoned, if only because I see little prospect of collecting the necessary data on the internal workings of all tuners. 

EMJB
